### ISSUE TITLE
Would like to have a private bastion

### ISSUE THREAD

**Issue by jessestephens1010** [oracle-terraform-modules/terraform-oci-oke#257]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I have a strict requirement in my tenancy to have no publicly accessible compute instances, even a bastion. So, I'd like to either be able to forget the bastion altogether and just have the operator or at the very least enable the bastion to have a private IP.

### New or Affected Resource(s)

oracle-terraform-modules/terraform-oci-oke
oracle-terraform-modules/bastion/oci

### Potential Terraform Configuration

oracle-terraform-modules > variables.tf
```hcl
variable "bastion_public" {
  description = "true to use public IP for bastion, false to use private IP"
  default     = true
  type        = boolean
}
```

oracle-terraform-modules > locals.tf 
```hcl
oci_base_bastion = {
    availability_domain   = var.availability_domains["bastion"]
    bastion_access        = var.bastion_access
    bastion_enabled       = var.bastion_enabled
    bastion_image_id      = var.bastion_image_id
    bastion_shape         = var.bastion_shape
    ***bastion_public        = var.bastion_public***
    bastion_upgrade       = var.bastion_package_upgrade
    ...
  }
```

oracle-terraform-modules/bastion/oci > compute.tf
```
resource "oci_core_instance" "bastion" {
  availability_domain = element(local.ad_names, (var.availability_domain - 1))
  compartment_id      = var.compartment_id
  freeform_tags       = var.tags

  create_vnic_details {
    ***assign_public_ip = var.bastion_public***
    display_name     = var.label_prefix == "none" ? "bastion-vnic" : "${var.label_prefix}-bastion-vnic"
    hostname_label   = "bastion"
    subnet_id        = oci_core_subnet.bastion[0].id
  }
  ...
```

### References

N/A

**Comment by hyder** [comment]
https://github.com/oracle-terraform-modules/terraform-oci-bastion/issues/23

**Comment by hyder** [comment]
closed in #332
